R/zchunk_LA115.RooftopPV.R

```diff
@@ -30,43 +28,125 @@ module_gcam.usa_LA115.RooftopPV_DISABLED <- function(command, ...) {
-    # There's also a `same_precursors_as(x)` you can use
-    # If no precursors (very rare) don't call `add_precursor` at all
-    tibble() %>%
-      add_title("descriptive title of data") %>%
-      add_units("units") %>%
-      add_comments("comments describing how data generated") %>%
-      add_comments("can be multiple lines") %>%
+    # Compute PV fixed charge rate
+    PV_FCR <- ( PV.DISCOUNT_RATE * ( 1 + PV.DISCOUNT_RATE ) ^ PV.LIFETIME ) /
+      ( ( PV.DISCOUNT_RATE + 1 ) ^ PV.LIFETIME - 1 )
+
+    # Get maximum residential capacity factor
+    filter(NREL_Res_PV_supply_curve, Relative_Cost == min(Relative_Cost))$MWh /
+      filter(NREL_Res_PV_supply_curve, Relative_Cost == min(Relative_Cost))$MW /
+      HOURS_PER_YEAR -> max_resid_capacity_factor
+
+    # Get maximum commerical capacity factor
+    filter(NREL_Com_PV_supply_curve, Rel_Cost == min(Rel_Cost))$GWH * CONV_BIL_MIL /
+      filter(NREL_Com_PV_supply_curve, Rel_Cost == min(Rel_Cost))$MW /
+      HOURS_PER_YEAR -> max_comm_capacity_factor
+
+    # Get minimum levelized electricity cost (LEC) for residential PV in 2005USD / KW
+    PV_resid_min_LEC_2005 <- (PV.RESID_INSTALLED_COST / PV.DERATING_FACTOR) * PV_FCR /
+      (max_resid_capacity_factor * HOURS_PER_YEAR) +
+      PV.RESID_OM / (max_resid_capacity_factor * HOURS_PER_YEAR)
+    # ... and in 1975USD / GJ
+    PV_resid_min_LEC_1975 <- PV_resid_min_LEC_2005 * gdp_deflator(1975, 2005) / CONV_KWH_GJ
+
+    # Get minimum levelized electricity cost (LEC) for commercial PV in 2005USD / KW
+    PV_comm_min_LEC_2005 <- (PV.COMM_INSTALLED_COST / PV.DERATING_FACTOR ) * PV_FCR /
+      (max_comm_capacity_factor * HOURS_PER_YEAR) +
+      PV.COMM_OM / (max_comm_capacity_factor * HOURS_PER_YEAR)
+    # ... and in 1975USD / GJ
+    PV_comm_min_LEC_1975 <- PV_comm_min_LEC_2005 * gdp_deflator(1975, 2005) / CONV_KWH_GJ
+
+    # COMPUTE SUPPLY CURVES
+
+    # Get residential supply curve
+    NREL_Res_PV_supply_curve %>%
+      mutate(p = Relative_Cost * PV_resid_min_LEC_1975,
+             generation = MWh / CONV_BIL_MIL) %>%  # << convert to GWh (like commercial)
+      rename(state_name = State) %>%
+      select(state_name, p, generation) ->
+      L115.pv_sc_res
+
+    # Get commercial supply curve
+    NREL_Com_PV_supply_curve %>%
+      mutate(p = Rel_Cost * PV_comm_min_LEC_1975,
+             generation = GWH) %>%
+      rename(state_name = State) %>%
+      select(state_name, p, generation) ->
+      L115.pv_sc_com
+
+    # Combine supply curves, add state abbrv., and rebase p
+    bind_rows(L115.pv_sc_res, L115.pv_sc_com) %>%
+      left_join_error_no_match(select(states_subregions, state_name, state),
+                               by = "state_name") %>%
+      arrange(state_name) %>%
+      select(-state_name, state, p, generation) %>%
+      mutate(p = p - min(p)) %>%
+      arrange(p) ->
+      # ^^ note: ordering of generation for matching p values will affect supply curve parameters (fitted below).
```

Comment:
Ugh

Reply:
Ok I'll catch up with you at some point to explain this issue (you around today?)

Reply:
Sounds good, thanks. I'm out today, unexpectedly, but in the rest of the week.

Reply:
I think it will be useful to mention the equation for the supply curve somewhere in the script (preferably upfront) and describe what parameters of that equation are being estimated here.

Reply:
@bpbond you around today or tomorrow to quickly discuss this??

Reply:
Tomorrow, sure, sounds good.